lib/CrashHandler/CrashHandler.cpp

```diff
@@ -248,120 +300,191 @@ void logCrashInfo(std::string_view context, int signal, siginfo_t* info,
-            } else {
-              // demangling has failed.
-              // in this case, append mangled version. still better than nothing
-              buffer.append(mangled);
-            }
-            // print offset into function
-            buffer.append(" (+0x").appendHexValue(offset, true).append(")");
+        appendAddress(frameBuffer, pc, base);
+
+        char mangled[512];
+        memset(&mangled[0], 0, sizeof(mangled));
+
+        // get symbol information (in mangled format)
+        unw_word_t offset = 0;
+        if (unw_get_proc_name(&cursor, &mangled[0], sizeof(mangled) - 1,
+                              &offset) == 0) {
+          // "mangled" buffer must have been null-terminated before, but it
+          // doesn't harm if we double-check it is null-terminated
+          mangled[sizeof(mangled) - 1] = '\0';
+
+          boost::core::scoped_demangled_name demangled(&mangled[0]);
+
+          if (demangled.get()) {
+            frameBuffer.append(demangled.get());
           } else {
-            // unable to retrieve symbol information
-            buffer.append("*no symbol name available for this frame");
+            // demangling has failed.
+            // in this case, append mangled version. still better than nothing
+            frameBuffer.append(mangled);
           }
+          // print offset into function
+          frameBuffer.append(" (+0x").appendHexValue(offset, true).append(")");
+        } else {
+          // unable to retrieve symbol information
+          frameBuffer.append("*no symbol name available for this frame");
+        }
 
-          LOG_TOPIC("308c3", INFO, arangodb::Logger::CRASH) << buffer.view();
+        frameBuffer.append("\n");
+
+        if (!safeAppend(frameBuffer.view())) {
+          break;  // Buffer is full
         }
-      } while (++frame < (maxFrames + skipFrames + 1) && unw_step(&cursor) > 0);
-      // flush logs as early as possible
-      arangodb::Logger::flush();
-    }
+      }
+    } while (++frame < (maxFrames + skipFrames + 1) && unw_step(&cursor) > 0);
   }
 
+  return totalWritten;
+#else
+  return 0;
 #endif
 } catch (...) {
   // we better not throw an exception from inside a signal handler
+  return 0;
+}
+
+/// @brief log a previously acquired backtrace from a buffer
+/// Note that this uses the Logger and thus should not be called directly
+/// in a signal handler. Use `acquireBacktrace` above to get the backtrace
+/// in the signal handler and then this function in the dedicated
+/// CrashHandler thread to do the actual logging!
+void logAcquiredBacktrace(char const* buffer, size_t bufferSize) try {
+  if (buffer == nullptr || bufferSize == 0) {
+    return;
+  }
+
+  // Split buffer content by lines and log each line
+  size_t pos = 0;
+  size_t lineStart = 0;
+
+  while (pos <= bufferSize) {
+    if (pos == bufferSize || buffer[pos] == '\n') {
+      if (pos > lineStart) {
+        std::string_view line(buffer + lineStart, pos - lineStart);
```

Comment:
If you change `buffer` into a `std::string_view`, you can just do
```suggestion
  for (auto const line : std::views::split(buffer, "\n")) {
```
instead, if you include `<ranges>`. :)